esp/esp/program/modules/handlers/volunteermanage.py

```diff
@@ -91,10 +100,67 @@ def volunteering(self, request, tl, one, two, module, extra, prog):
         else:
             form = VolunteerRequestForm(program=prog)
 
-        context['form'] = form
+        context['shift_form'] = form
+        if 'import_request_form' not in context:
+            context['import_request_form'] = VolunteerImportForm()
         context['requests'] = self.program.getVolunteerRequests()
         return render_to_response('program/modules/volunteermanage/main.html', request, context)
 
+    def volunteer_import(self, request, tl, one, two, module, extra, prog):
+        context = {}
+        response = None
+
+        import_mode = 'preview'
```

Comment:
If `import_mode` can only have two values, just make it a boolean, e.g. `has_confirmed = request.POST.get('import_confirm') == 'yes'`.